client/src/components/ImageGrid.jsx

```diff
@@ -1,75 +1,80 @@
-import React from 'react';
+import { useEffect, useState } from 'react';
 import { Gallery } from 'react-grid-gallery';
 import '../styles/global.css';
 
-class ImageGrid extends React.Component{
-	constructor(props){
-		super(props)
-		this.state={
-			images:[]
-		};
-
-		this.handleClick = this.handleClick.bind(this)
-	}
+function ImageGrid(props) {
+	const [images, setImages] = useState([]);
 
-	componentDidMount(){
+	useEffect(() => {
 		let jsonData = {
-			"tag": "*"
+			tag: '*',
 		};
 
-		fetch("http://localhost:5000/queryimages",{
+		fetch('http://localhost:5000/queryimages', {
 			method: 'POST',
-			mode:"cors",
-			headers: {'Content-Type':'application/json'},
-			body: JSON.stringify(jsonData)
-		}).then(response => response.json())
-		.then(data => {
-			let imgs = [];
-			data.forEach(img => {
-				let tags = [];
-				img["tags"].forEach(tag => {
-					tags.push({"value":tag ,"title":tag});
-				});
-				imgs.push({
-					"src":img["url"],
-					"tags":tags
-				});
-			});
-			this.setState({images:imgs});
+			mode: 'cors',
+			headers: { 'Content-Type': 'application/json' },
+			body: JSON.stringify(jsonData),
 		})
-		.catch(error => console.log('error', error));
-	}
+			.then((response) => response.json())
+			.then((data) => {
+				let imgs = [];
+				data.forEach((img) => {
+					let selectedTags = [];
+					props.selectedTags.forEach((tag) => {
+						selectedTags.push(tag['tag']);
+					});
+					let tags = [];
+					img['tags'].forEach((tag) => {
+						tags.push({ value: tag, title: tag });
+					});
+					let tagvalues = [];
+					tags.forEach((tag) => {
+						tagvalues.push(tag['value']);
+					});
+					if (
+						tagvalues.filter((value) => selectedTags.includes(value)).length !==
+						0
+					) {
+						imgs.push({
+							src: img['url'],
+							tags: tags,
+						});
+					}
+				});
+				setImages(imgs);
+			})
+			.catch((error) => console.log('error', error));
+	}, [props.selectedTags]);
```

Comment:
We should make the db handle the query and not the client, so the client isn't always pulling every image and filtering. Since the API doesn't currently support multiple tags like TagFilter allows, we would need to modify the /queryimages endpoint to handle lists of tags. (we may also want to put a limit on the number of images returned from the API)